test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/SourceTelemetryTests.cs

```diff
@@ -405,6 +421,53 @@ public void VerifyProtocolDiagnosticTotals()
             telemetry[ProtocolDuration].Should().Be(3.0);
         }
 
+
+        [Fact]
+        public void LocalAndHttpSources_WithAllHttps_IsHttpsIsTrue()
+        {
+            var sources = new List<PackageSource>()
+            {
+                new PackageSource(@"\\share\packages"),
+                new PackageSource("https://nugettest.org/v3/index.JSON"),
+                new PackageSource("https://tempuri.local/index.json"),
+                new PackageSource("https://nuget.org/v3/index.JSON")
+            };
+
+            var summary = SourceTelemetry.GetRestoreSourceSummaryEvent(Parent, sources, ProtocolDiagnosticTotals);
+            var summaryStrings = GetValuesAsStrings(summary);
+            var summaryInts = GetValuesAsInts(summary);
+            var summaryBools = GetValuesAsBools(summary);
+
+            summaryInts[NumLocalFeeds].Should().Be(1);
+            summaryInts[NumHTTPv2Feeds].Should().Be(0);
+            summaryInts[NumHTTPv3Feeds].Should().Be(3);
+            summaryInts[NumHTTPSFeeds].Should().Be(3);
+            summaryStrings[ParentId].Should().Be(Parent.ToString());
+        }
+
+        [Fact]
+        public void LocalAndHttpSources_WithAnyNonHttps_IsHttpsIsFalse()
+        {
+            var sources = new List<PackageSource>()
+            {
+                new PackageSource(@"\\share\packages"),
+                new PackageSource("https://nugettest.org/v3/index.JSON"),
+                new PackageSource("http://tempuri.local/index.json"),
+                new PackageSource("https://nuget.org/v3/index.JSON")
+            };
+
+            var summary = SourceTelemetry.GetRestoreSourceSummaryEvent(Parent, sources, ProtocolDiagnosticTotals);
+            var summaryStrings = GetValuesAsStrings(summary);
+            var summaryInts = GetValuesAsInts(summary);
+            var summaryBools = GetValuesAsBools(summary);
+
+            summaryInts[NumLocalFeeds].Should().Be(1);
+            summaryInts[NumHTTPv2Feeds].Should().Be(0);
+            summaryInts[NumHTTPv3Feeds].Should().Be(3);
+            summaryInts[NumHTTPSFeeds].Should().Be(2);
+            summaryStrings[ParentId].Should().Be(Parent.ToString());
```

Comment:
I think it would be useful to assert that all feeds are accounted. For example (pseudocode):

`Assert.True( TotalFeeds == HttpsFeedsCount + HttpV2FeedsCount + HttpV3FeedsCount )`

Reply:
See if what I added meets your expectation